src/client/ua_client_connect.c

```diff
@@ -1291,12 +1317,40 @@ __Client_networkCallback(UA_ConnectionManager *cm, uintptr_t connectionId,
     UA_UNLOCK(&client->clientMutex);
 }
 
+static void closeListeningReverseConnectSocket(UA_Client *client) {
+    UA_LOCK(&client->clientMutex);
```

Comment:
You are unlocking below just to lock again here.
Not needed.

Reply:
Done